plugins/outputs/otlp/otlp.go

```diff
@@ -0,0 +1,379 @@
+		IsMonotonic:            true,
+		AggregationTemporality: otlpmetricpb.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+		DataPoints:             []*otlpmetricpb.IntDataPoint{integer},
+	}
+}
+
+func monotonicDoublePoint(labels []*otlpcommonpb.StringKeyValue, start, end int64, value float64) *otlpmetricpb.DoubleSum {
+	double := &otlpmetricpb.DoubleDataPoint{
+		Labels:            labels,
+		StartTimeUnixNano: getNanos(start),
+		TimeUnixNano:      getNanos(end),
+		Value:             value,
+	}
+	return &otlpmetricpb.DoubleSum{
+		IsMonotonic:            true,
+		AggregationTemporality: otlpmetricpb.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+		DataPoints:             []*otlpmetricpb.DoubleDataPoint{double},
+	}
+}
+
+func protoLabel(tag *telegraf.Tag) *otlpcommonpb.KeyValue {
+	return &otlpcommonpb.KeyValue{
+		Key: tag.Key,
+		Value: &otlpcommonpb.AnyValue{
+			Value: &otlpcommonpb.AnyValue_StringValue{
+				StringValue: tag.Value,
+			},
+		},
+	}
+}
+
+func protoStringLabel(tag *telegraf.Tag) *otlpcommonpb.StringKeyValue {
+	return &otlpcommonpb.StringKeyValue{
+		Key:   tag.Key,
+		Value: tag.Value,
+	}
+}
+
+func protoResourceAttributes(tags []*telegraf.Tag) []*otlpcommonpb.KeyValue {
+	ret := make([]*otlpcommonpb.KeyValue, len(tags))
+	for i := range tags {
+		ret[i] = protoLabel(tags[i])
+	}
+	return ret
+}
+
+func protoStringLabels(tags []*telegraf.Tag) []*otlpcommonpb.StringKeyValue {
+	ret := make([]*otlpcommonpb.StringKeyValue, len(tags))
+	for i := range tags {
+		ret[i] = protoStringLabel(tags[i])
+	}
+	return ret
+}
+
+func (o *OTLP) protoTimeseries(m telegraf.Metric, f *telegraf.Field) (*otlpmetricpb.ResourceMetrics, *otlpmetricpb.Metric) {
+	metric := &otlpmetricpb.Metric{
+		Name:        fmt.Sprintf("%s.%s", m.Name(), f.Key),
+		Description: "", // TODO
+		Unit:        "", // TODO
+	}
+	return &otlpmetricpb.ResourceMetrics{
+		Resource: &otlpresourcepb.Resource{
+			Attributes: protoResourceAttributes(o.resourceTags),
+		},
+		InstrumentationLibraryMetrics: []*otlpmetricpb.InstrumentationLibraryMetrics{
+			{
+				InstrumentationLibrary: &otlpcommonpb.InstrumentationLibrary{
+					Name:    instrumentationLibraryName,
+					Version: internal.Version(),
+				},
+				Metrics: []*otlpmetricpb.Metric{metric},
+			},
+		},
+	}, metric
+}
+
+func intGauge(labels []*otlpcommonpb.StringKeyValue, ts int64, value int64) *otlpmetricpb.IntGauge {
+	integer := &otlpmetricpb.IntDataPoint{
+		Labels:       labels,
+		TimeUnixNano: uint64(ts),
+		Value:        value,
+	}
+	return &otlpmetricpb.IntGauge{
+		DataPoints: []*otlpmetricpb.IntDataPoint{integer},
+	}
+}
+
+func doubleGauge(labels []*otlpcommonpb.StringKeyValue, ts int64, value float64) *otlpmetricpb.DoubleGauge {
+	double := &otlpmetricpb.DoubleDataPoint{
+		Labels:       labels,
+		TimeUnixNano: uint64(ts),
+		Value:        value,
+	}
+	return &otlpmetricpb.DoubleGauge{
+		DataPoints: []*otlpmetricpb.DoubleDataPoint{double},
+	}
+}
```

Comment:
It might be worth to move those to a separate file to reduce the clutter.

Reply:
Done, moved this to `transform.go`